### ISSUE TITLE
XHR Request

### ISSUE THREAD

**Issue by pkrawc** [jodyanna/portfolio_v2#2]
There's a much easier API to use for data fetching. The `fetch` api. It's supported across all evergreen browsers, to use it in node you'll need a polyfill like `node-fetch`.

It's a function that builds on top of the http request api so instead of the following.

```js
export const ajaxRequest = (url) => {
  return new Promise((resolve) => {
    const xhttp = new XMLHttpRequest();
    xhttp.onreadystatechange = function() {
      if (this.readyState === 4 && this.status === 200) resolve(JSON.parse(this.responseText))
    };
    xhttp.open("GET", url, true);
    xhttp.send();
    setTimeout(() => {}, 1000)
  })
}
```

You can now write something like

```js
const data = Promise(resolve => fetch(url).then(
  // The response itself is a blob that you need to transform into json, just like the above.
  response => response.json().then(body => resolve(body))
))
```

This API combined with `async/await` makes including data into your components a breeze.

```js
// With the async/await style. Also supported by all modern browsers out of the box.
// The await keyword indicates that the return value in front of it is a promise and waits for it to resolve.
const data = await fetch("/posts/123").then(res => res.json())
```

This should probably be used with a `try/catch` block to make sure you're catching errors correctly

**Comment by jodyanna** [comment]
Yeah, I had a feeling that the XMLHttpRequest object was getting dated,
thanks for clearing that up for me. I'm working on a full stack CRUD app at
the moment using Express for the API and React for the client side stuff. I
decided to use fetch for the HTTP requests and, you are right, it is much
easier.

Funny story! So when I was first getting into coding last year I decided to
take a couple community college courses to get my feet wet and have some
fun. I took their JavaScript course and we were learning like early-2010's
conventions in 2019. Declaring variables with var, old AJAX requests,
etc... ES6 has been out for a couple years and we were learning ES5. I
guess the knowledge is good if I ever come across older code bases, either
way, kind of funny. Damn it Waubonsee!

Anyways, thanks again for the advice!
Take care
